pkg/reporting/cookbooks_test.go

```diff
@@ -257,12 +257,89 @@ func TestCookbooks_ListCookbooksErrors(t *testing.T) {
 	assert.Nil(t, c)
 }
 
+func TestCookbooks_ListAvailableVersionsError(t *testing.T) {
+	cookbookList := chef.CookbookListResult{}
+	c, err := subject.NewCookbooks(
+		newMockCookbook(cookbookList, errors.New("list error"), nil),
+		nil,
+		false,
+	)
+	assert.Nil(t, c)
+	assert.EqualError(t, err, "unable to retrieve cookbooks: list error")
+}
+
+func TestCookbooks_NoneAvailable(t *testing.T) {
+
+	c, err := subject.NewCookbooks(
+		newMockCookbook(chef.CookbookListResult{}, nil, nil),
+		makeMockSearch(mockedEmptyNodesSearchRows(), nil),
+		false,
+	)
+	assert.Nil(t, err)
+	if assert.NotNil(t, c) {
+		assert.Equal(t, 0, c.TotalCookbooks)
+	}
+}
+
 // Given a failure in downloading a cookbook, verify
 // the result set is as expected
 func TestCookbooks_DownloadErrors(t *testing.T) {
+	savedPath := setupBinstubsDir()
+	defer os.Setenv("PATH", savedPath)
+
+	cookbookList := chef.CookbookListResult{
+		"foo": chef.CookbookVersions{
+			Versions: []chef.CookbookVersion{
+				chef.CookbookVersion{Version: "0.1.0"},
+			},
+		},
+	}
+	c, err := subject.NewCookbooks(
+		newMockCookbook(cookbookList, nil, errors.New("download error")),
+		makeMockSearch(mockedNodesSearchRows(), nil),
+		false,
+	)
+	assert.Nil(t, err)
+	if assert.NotNil(t, c) {
+		assert.Equal(t, 1, c.TotalCookbooks)
+		if assert.Equal(t, 1, len(c.Records)) {
+			for _, rec := range c.Records {
+				assert.EqualError(t, rec.DownloadError, "unable to download cookbook foo: download error")
+				assert.NoError(t, rec.UsageLookupError, "unexpected UsageLookupError")
+				assert.NoError(t, rec.CookstyleError, "unexpected UsageLookupError")
+			}
+		}
+	}
 }
 
 // Given a failure in fetching node usage for a cookbook,
 // the result set is as expected
 func TestCookbooks_UsageStatErrors(t *testing.T) {
+	savedPath := setupBinstubsDir()
+	defer os.Setenv("PATH", savedPath)
+
+	cookbookList := chef.CookbookListResult{
+		"foo": chef.CookbookVersions{
+			Versions: []chef.CookbookVersion{
+				chef.CookbookVersion{Version: "0.1.0"},
+			},
+		},
+	}
+	c, err := subject.NewCookbooks(
+		newMockCookbook(cookbookList, nil, nil),
+		makeMockSearch(mockedNodesSearchRows(), errors.New("lookup error")),
+		false,
+	)
+	assert.Nil(t, err)
+	if assert.NotNil(t, c) {
+		assert.Equal(t, 1, c.TotalCookbooks)
+		if assert.Equal(t, 1, len(c.Records)) {
```

Comment:
I think this should be `0` since we are returning an error? 🤔

Reply:
I think 1 is right here - we processed one cookbook into `Records`, and it has an error.

Reply:
Got this fixed by saying "only return unused cookbooks" since I reverted my change to `nodesUsingCookbookVersion` - it now goes back to returning `nil, err` if there is a problem querying information from the Chef Server and that means the cookbook looks like it isn't assigned to any nodes.

Open to discussion on this (created a slack post where we can discuss)